stdlib/SparseArrays/src/linalg.jl

```diff
@@ -1359,7 +1359,7 @@ function copyinds!(C::AbstractSparseMatrixCSC, A::AbstractSparseMatrixCSC)
 end
 
 # multiply by diagonal matrix as vector
-function mul!(C::AbstractSparseMatrixCSC, A::AbstractSparseMatrixCSC, D::Diagonal{T, <:Vector}) where T
+function mul!(C::AbstractSparseMatrixCSC, A::AbstractSparseMatrixCSC, D::Diagonal)
     m, n = size(A)
     b    = D.diag
```

Comment:
Now that `b` can be any vector, would it make sense to call `require_one_based_indexing`?

```suggestion
     b    = D.diag
     require_one_based_indexing(b)
```

Reply:
It wouldn't hurt, but it's unnecessary, because the `.diag` object has to be one-based by definition of `Diagonal`, see the very top of `diagonal.jl`.

Reply:
Oh, I see. Thanks. I missed that.